In [9]:
from prettytable import PrettyTable

import json
def load_simple_json(filename):
    with open(filename, 'r') as f:
        return json.load(f)
    
words=load_simple_json('de/occupation_all.json')#{masculine:[[],[feminine]]}
words_revert={words[i][1]:i for i in words}#feminine:masculine
#neutral=load_simple_json('de/neutral_new.json')

no_page_m=load_simple_json('de/wiki/m_no_page.json')
no_page_f=load_simple_json('de/wiki/f_no_page.json')
#no_page_n=load_simple_json('de/wiki/n_no_page.json')
page_m=load_simple_json('de/wiki/m_page.json')
page_f=load_simple_json('de/wiki/f_page.json')
#page_n=load_simple_json('de/wiki/n_page.json')
m_page_levenshtein=load_simple_json('de/wiki/m_page_levenshtein.json')
#n_page_levenshtein=load_simple_json('de/wiki/n_page_levenshtein.json')


page_f_validated=load_simple_json('de/wiki/f_page_validated.json')
page_f_ambigious=load_simple_json('de/wiki/f_page_ambigious.json')
page_f_other=load_simple_json('de/wiki/f_page_other.json')  
    
page_m_validated=load_simple_json('de/wiki/m_page_validated.json') 
page_m_ambigious=load_simple_json('de/wiki/m_page_ambigious.json')
page_m_other=load_simple_json('de/wiki/m_page_other.json')

#page_n_validated=load_simple_json('de/wiki/n_page_validated.json')
#page_n_ambigious=load_simple_json('de/wiki/n_page_ambigious.json')
#page_n_other=load_simple_json('de/wiki/n_page_other.json') 

m_links_to_feminine=load_simple_json('de/wiki/m_links_to_feminine.json')
m_links_to_smth=load_simple_json('de/wiki/m_links_to_smth.json')
f_links_to_mascuilne=load_simple_json('de/wiki/f_links_to_mascuilne.json')
f_links_to_smth=load_simple_json('de/wiki/f_links_to_smth.json')
f_links_to_mascuilne_with_link=load_simple_json('de/wiki/f_links_to_mascuilne_with_link.json')
#redirection_n=load_simple_json('de/wiki/n_redirects.json')
f_links_to_mascuilne_lev=load_simple_json('de/wiki/f_links_to_mascuilne_lev.json')

t = PrettyTable(['-de-',"Occup.", 'Wiki pages', "Validated pages","Ambigious pages","Redirects",
                 "Redir. to oposit gender"])
t.add_row(['Masculine',len(words), len(page_m)+len(m_page_levenshtein), len(page_m_validated), len(page_m_ambigious),
           len(m_links_to_feminine)+len(m_links_to_smth), len(m_links_to_feminine)])
t.add_row(['Feminine',len(words_revert), len(page_f), len(page_f_validated), len(page_f_ambigious),
           len(f_links_to_mascuilne)+len(f_links_to_smth)+len(f_links_to_mascuilne_lev),
           len(f_links_to_mascuilne)+len(f_links_to_mascuilne_lev)])
print t

t2 = PrettyTable(['-de-',"Occup.", 'Wiki pages', "Redirects", "No page"])
t2.add_row(['Masculine',len(words), len(page_m)+len(m_page_levenshtein), len(m_links_to_feminine)+len(m_links_to_smth),
            len(no_page_m)-len(m_page_levenshtein)])
t2.add_row(['Feminine',len(words_revert), len(page_f),
            len(f_links_to_mascuilne)+len(f_links_to_smth)+len(f_links_to_mascuilne_lev),
            len(no_page_f)-len(f_links_to_mascuilne_lev)])
print t2
print
t3 = PrettyTable(['-de-','Wiki pages', "Validated pages","Ambigious pages","Not validated pages"])
t3.add_row(['Masculine', len(page_m)+len(m_page_levenshtein), len(page_m_validated)+len(m_page_levenshtein), len(page_m_ambigious),len(page_m_other)])
t3.add_row(['Feminine', len(page_f), len(page_f_validated), len(page_f_ambigious),len(page_f_other)])
print t3
print
t4 = PrettyTable(['--de--',"Redirects","Redirects to oposit gender", "Other redirects"])
t4.add_row(['Masculine',len(m_links_to_feminine)+len(m_links_to_smth),
            len(m_links_to_feminine),len(m_links_to_smth)])
t4.add_row(['Feminine',len(f_links_to_mascuilne)+len(f_links_to_smth)+len(f_links_to_mascuilne_lev),
            len(f_links_to_mascuilne)+len(f_links_to_mascuilne_lev),len(f_links_to_smth)])
print t4

+-----------+--------+------------+-----------------+-----------------+-----------+-------------------------+
|    -de-   | Occup. | Wiki pages | Validated pages | Ambigious pages | Redirects | Redir. to oposit gender |
+-----------+--------+------------+-----------------+-----------------+-----------+-------------------------+
| Masculine |  4239  |    909     |       732       |       114       |    499    |            4            |
|  Feminine |  4239  |     15     |        8        |        5        |    286    |           192           |
+-----------+--------+------------+-----------------+-----------------+-----------+-------------------------+
+-----------+--------+------------+-----------+---------+
|    -de-   | Occup. | Wiki pages | Redirects | No page |
+-----------+--------+------------+-----------+---------+
| Masculine |  4239  |    909     |    499    |   2831  |
|  Feminine |  4239  |     15     |    286    |   3939  |
+-----------+--------+------------+-----------+---

In [11]:
from prettytable import PrettyTable
from tabulate import tabulate

import json
def load_simple_json(filename):
    with open(filename, 'r') as f:
        return json.load(f)


headers=["Occupation", 'Wiki pages', "Redirects","No wiki page or redir","All"]
t = PrettyTable(headers)


table=[]
for l in ["de","fr","ru"]:
    m_no_page=load_simple_json(l+'/wiki/m_no_page.json')
    m_redirection=load_simple_json(l+'/wiki/m_redirection.json')
    m_page=load_simple_json(l+'/wiki/m_page.json')
    words=len(m_no_page)+len(m_page)+ len(m_redirection)
    if l=="de":
        words_data=load_simple_json('de/occupation_all.json')
        words=len(words_data)
        m_page_levenshtein=load_simple_json(l+'/wiki/m_page_levenshtein.json')
        f_links_levenshtein=load_simple_json(l+'/wiki/f_links_to_mascuilne_lev.json')
        t.add_row(['{0}: Masculine'.format(l), len(m_page)+len(m_page_levenshtein),
                   len(m_redirection),len(m_no_page)-len(m_page_levenshtein),words])
        f_no_page=load_simple_json(l+'/wiki/f_no_page.json')
        f_redirection=load_simple_json(l+'/wiki/f_redirection.json')
        f_page=load_simple_json(l+'/wiki/f_page.json')
        t.add_row(['{0}:  Feminine'.format(l),len(f_page),len(f_redirection),len(f_no_page)-len(f_links_levenshtein), words])
        table.append(['{0}: Masculine'.format(l), len(m_page)+len(m_page_levenshtein),
                   len(m_redirection),len(m_no_page)-len(m_page_levenshtein),words])
        table.append(['{0}:  Feminine'.format(l), len(f_page),len(f_redirection),len(f_no_page),words])
    else:
        t.add_row(['{0}: Masculine'.format(l),len(m_page),len(m_redirection),len(m_no_page),words])
        table.append(['{0}: Masculine'.format(l),len(m_page),len(m_redirection),len(m_no_page),words])
print t

print tabulate(table, headers=headers, tablefmt="latex")#, numalign="left")



+---------------+------------+-----------+-----------------------+------+
|   Occupation  | Wiki pages | Redirects | No wiki page or redir | All  |
+---------------+------------+-----------+-----------------------+------+
| de: Masculine |    909     |    499    |          2831         | 4239 |
| de:  Feminine |     15     |    286    |          3939         | 4239 |
| fr: Masculine |     92     |     47    |          113          | 252  |
| ru: Masculine |    227     |    147    |          7296         | 7670 |
+---------------+------------+-----------+-----------------------+------+
\begin{tabular}{lrrrr}
\hline
 Occupation    &   Wiki pages &   Redirects &   No wiki page or redir &   All \\
\hline
 de: Masculine &          909 &         499 &                    2831 &  4239 \\
 de:  Feminine &           15 &         286 &                    3943 &  4239 \\
 fr: Masculine &           92 &          47 &                     113 &   252 \\
 ru: Masculine &          227 &         147 &  

\begin{tabular}{llll|l}
\hline
 Occupation    &   Wiki pages &   Redirects &   No wiki page or redir &   All \\
\hline
 de: Masculine &          909 &         499 &                    2831 &  4239 \\
 de:  Feminine &           15 &         286 &                    3943 &  4239 \\
 fr: Masculine &           92 &          47 &                     113 &   252 \\
 ru: Masculine &          227 &         147 &                    7296 &  7670 \\
\hline
\end{tabular}